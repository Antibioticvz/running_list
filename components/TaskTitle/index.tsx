export { default as TaskTitle, TaskStatus } from './TaskTitle';

