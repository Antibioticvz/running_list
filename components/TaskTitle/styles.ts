const styles = {
    container: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    title: {
        width: '88%',
        // backgroundColor: 'skyblue',
    },
    titleText: {
        marginRight: '2%',
        textAlign: 'right',
    },
    icon: {
        width: '12%', 
        // backgroundColor: 'powderblue',
        alignItems: 'center',
    },
}

export default styles