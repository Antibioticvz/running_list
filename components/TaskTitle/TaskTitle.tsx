import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View as RView } from 'react-native'
import { Text } from 'react-native-ui-lib'
import platformIconPrefix from '../../helpers/platformIconPrefix'
import styles from './styles'

export enum TaskStatus {
    active = 'ACTIVE',
    done = 'DONE',
    closed = 'CLOSED'
}
interface Props {
    title: string,
    status: TaskStatus,
}

const TaskTitle: React.FC<Props> = ({title, status}) => {
    const getStatusIcon = (status: TaskStatus) =>{
        switch (status) {
            case TaskStatus.active:
                return <Ionicons name={`${platformIconPrefix()}-radio-button-off`} size={40} color="blue" />
            case TaskStatus.done:
                return <Ionicons name={`${platformIconPrefix()}-checkmark-circle-outline`} size={40} color="green" />
            case TaskStatus.closed:
                return <Ionicons name={`${platformIconPrefix()}-close-circle-outline`} size={40} color="red" />
            default:
                return null
        }
    }
    return (
            <RView style={styles.container}>
                <RView style={styles.icon}>
                    { getStatusIcon(status) }
                </RView>
                <RView style={styles.title} >
                    <Text text40 style={styles.titleText}>
                        {title}
                    </Text>
                </RView>
            </RView>
    )
}

export default TaskTitle
