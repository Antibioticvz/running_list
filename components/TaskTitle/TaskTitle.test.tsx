import React from 'react'
import 'react-native'
import { NativeModules } from 'react-native'
import renderer from 'react-test-renderer'
import { TaskStatus, TaskTitle } from '../../components/TaskTitle'

NativeModules.StatusBarManager = {getHeight: jest.fn()}

describe('TaskTitle test', () => {
  it('Task Title', () => {
    const description = renderer.create(
    <TaskTitle {...{
        status: TaskStatus.done,
        title: 'Done',
    }}
    />).toJSON()
    expect(description).toMatchSnapshot()
  })

  it('Task Title', () => {
    const description = renderer.create(
    <TaskTitle {...{
        status: TaskStatus.active,
        title: 'Active',
    }}
    />).toJSON()
    expect(description).toMatchSnapshot()
  })

  it('Task Title', () => {
    const description = renderer.create(
    <TaskTitle {...{
        status: TaskStatus.closed,
        title: 'Closed',
    }}
    />).toJSON()
    expect(description).toMatchSnapshot()
  })
})