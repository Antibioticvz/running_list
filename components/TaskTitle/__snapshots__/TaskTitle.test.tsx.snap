// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTitle test Task Title 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "12%",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Object {
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "textAlign": "left",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 28,
            "fontWeight": "300",
            "lineHeight": 33,
          },
          undefined,
          Object {},
          undefined,
          Object {
            "marginRight": "2%",
            "textAlign": "right",
          },
        ]
      }
      text40={true}
    >
      Done
    </Text>
  </View>
</View>
`;

exports[`TaskTitle test Task Title 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "12%",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Object {
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "textAlign": "left",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 28,
            "fontWeight": "300",
            "lineHeight": 33,
          },
          undefined,
          Object {},
          undefined,
          Object {
            "marginRight": "2%",
            "textAlign": "right",
          },
        ]
      }
      text40={true}
    >
      Active
    </Text>
  </View>
</View>
`;

exports[`TaskTitle test Task Title 3`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "12%",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Object {
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "textAlign": "left",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 28,
            "fontWeight": "300",
            "lineHeight": 33,
          },
          undefined,
          Object {},
          undefined,
          Object {
            "marginRight": "2%",
            "textAlign": "right",
          },
        ]
      }
      text40={true}
    >
      Closed
    </Text>
  </View>
</View>
`;
